src/test/java/org/folio/circulation/domain/LoanTest.java

```diff
@@ -0,0 +1,119 @@
+package org.folio.circulation.domain;
+
+import static api.support.matchers.JsonObjectMatcher.hasJsonPath;
+import static api.support.matchers.TextDateTimeMatcher.isEquivalentTo;
+import static org.folio.circulation.domain.ItemStatus.AGED_TO_LOST;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.allOf;
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.assertFalse;
+
+import org.folio.circulation.domain.policy.Period;
+import org.folio.circulation.domain.policy.lostitem.LostItemPolicy;
+import org.joda.time.DateTime;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import api.support.builders.ItemBuilder;
+import api.support.builders.LoanBuilder;
+import api.support.builders.LostItemFeePolicyBuilder;
+import junitparams.JUnitParamsRunner;
+import junitparams.Parameters;
+import junitparams.converters.Nullable;
+
+@RunWith(JUnitParamsRunner.class)
+public class LoanTest {
```

Comment:
I'm glad that you added tests for this behaviour.

I wonder if the loan is the appropriate place for this determination. Instead of calculating these properties within the loan itself, could they be calculated elsewhere and passed in as parameters?

Reply:
They are calculated by lost item policy, in a fact. 

Why do you think it is important to decouple this logic?

Reply:
> They are calculated by lost item policy, in a fact.

Yet, the test for the calculation is within the loan. I think that is a symptom of a design challenge.

> Why do you think it is important to decouple this logic?

I don't think the loan should be aware of the policies. To me, the loan is there to ensure that state transitions are done in a coherent way and the specific rules about the triggering of those transitions lives elsewhere. Otherwise, classes like loans tend to get big and have lots of responsibilities.

Broader than that, I think there is a general design challenge in FOLIO, in that we really shouldn't be storing intermediate state like this on the loan itself. However we aren't going to get away from that in this change, so I wasn't going to raise it at all.

Does that make sense?

Reply:
Ok, refactored